src/matrixrtc/NewMembershipManager.ts

```diff
@@ -172,7 +202,7 @@ export class MembershipManager implements IMembershipManager {
         this.focusActive = focusActive;
         this.leavePromiseDefer = undefined;
 
-        this.scheduler.resetState();
+        this.state = MembershipManager.defaultState;
 
         this.scheduler
```

Comment:
I suspect the `.then()` needs to go before `.catch()`.

Reply:
ah es lint is what is making this a hard error?